app/javascript/controllers/country_state_controller.js

```diff
@@ -0,0 +1,46 @@
+import { Controller } from "@hotwired/stimulus";
+import { locationHelper } from "../helpers/locationHelper.js";
+
+export default class extends Controller {
+  static targets = ["country", "state"];
+
+  connect() {
+    this.selectedCountry = this.countryTarget.dataset.initialValue;
+    this.selectedState = this.stateTarget.dataset.initialValue;
+
+    this.setCountryDropdown();
+    this.setStateDropdown();
+  }
```

Comment:
I'd be curious to see if we can try this approach using hotwire to avoid writing JS - https://www.youtube.com/watch?v=ReZS6wfh3lo

I'am a pretty big advocate for less JS to maintain.

Reply:
Totally understandable. At first, I thought about using a gem and turbo requests to solve the problem, like in the video you shared. However, after careful consideration, I decided not to add the gem. Instead, I chose to handle the task directly on the client side.

The reason for this decision is that managing the logic on the client side can be beneficial. It reduces the burden on the server and avoids unnecessary GET requests for better overall performance and responsiveness.

Reply:
Such a good use of hotwire! I think this would be a great V2 @edwinthinks. Im kind of inclined to move forward with the PR as is since it works as expected and is not too much js to maintain

Reply:
Thank you for detailed explanation. I love these convos too! 

> Every bit of code added is weight and come with a cost.

I understand your perspective. That is right. 

> To summarize, I think we should use the easiest solution possible which I believe is a select form with options being countries

Since the PR is done and all working, I suggest keeping this as is. This ensures that the data collected is more accurate. Also users spend less time filling in boring information.

I am trying to look at the topic from user's perspective. If I were using this app and could only select the country, I might wonder why the state dropdown was excluded. 

However, I am totally open about what ever we decide to do. Let me know what I should do.

Reply:
@egemen-dev @edwinthinks Yeah, Im in favor of the hotwire solution but also ok with submitting this PR (with the Mexico changes) as long as we can commit to a V2 in the short term - happy to pair on that!

Reply:
@nsiwnf sounds like a excellent issue! 

I'am also good with merging this in with the understanding we will revisit. Let's keep the momentum going!